### PR TITLE
CPR-436 Add initial reclustering check

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/ReclusterService.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonKeyEntity
+import uk.gov.justice.digital.hmpps.personrecord.model.types.UUIDStatusType
+import uk.gov.justice.digital.hmpps.personrecord.service.EventKeys
+import uk.gov.justice.digital.hmpps.personrecord.service.RetryExecutor.ENTITY_RETRY_EXCEPTIONS
+import uk.gov.justice.digital.hmpps.personrecord.service.RetryExecutor.runWithRetry
+import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.CPR_UUID_RECLUSTER_NEEDS_ATTENTION
+
+@Component
+class ReclusterService(
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/ReclusterService.kt:16]: Not adding to any update flows yet as I don't see the point until developed. Still can write test for it regardless in isolation

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/PersonKeyService.kt [hunk 2]
```diff
@@ -9,7 +9,6 @@ import uk.gov.justice.digital.hmpps.personrecord.service.EventKeys
 import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
 import uk.gov.justice.digital.hmpps.personrecord.service.search.MatchResult
 import uk.gov.justice.digital.hmpps.personrecord.service.search.SearchService
-import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/PersonKeyService.kt:12]: ❤️

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/PersonKeyService.kt:12]: Can this also be removed from the DeletionService?

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**